@broskoski Artist alt

1aurabrown opened the pull request:
so, this is basically identical to my last pr except non-breaking. the `blurb` field is still a string here, with no field for credit to the gallery if the fallback is used. would like to add that but perhaps in a separate pr so that if there's anything that needs to be done on client apps we can coordinate.

mzikherman commented on schema/artist/index.js:
```diff
@@ -103,11 +111,38 @@ const ArtistType = new GraphQLObjectType({
       deathday: {
         type: GraphQLString,
       },
+      biography: {
```
Cool, I didn't realize those were articles.

mzikherman commented on schema/artist/index.js:
```diff
@@ -103,11 +111,38 @@ const ArtistType = new GraphQLObjectType({
       },
       meta: Meta,
-      blurb: markdown(),
+      blurb: {
```
makes sense, if there's a blurb use it, otherwise return the biography for the selected partner artist.

The blurb on an artist can be markdown, but the bio on partner artist is just plain text?

broskoski replied:
Also wondering about this!

1aurabrown replied:
correct

mzikherman commented on schema/artist/index.js:
```diff
@@ -131,8 +166,6 @@ const ArtistType = new GraphQLObjectType({
               published_artworks_count),
             follows: numeral(({ follow_count }) =>
               follow_count),
-            auction_lots: numeral(({ auction_lots_count }) =>
```
👋

mzikherman commented on schema/artist/statuses.js:
```diff
@@ -10,61 +10,72 @@ const ArtistStatusesType = new GraphQLObjectType({
   fields: {
     artworks: {
       type: GraphQLBoolean,
+      resolve: ({ published_artworks_count }) => !!published_artworks_count,
```
what's it mean for `!!` on an integer? Should it just be `published_artworks_count > 0` (since it can never be nil)

1aurabrown replied:
same thing, can change it if you like it better

On Fri, Jun 10, 2016 at 6:07 PM, mzikherman <notifications@github.com>
wrote:

> In schema/artist/statuses.js
> <https://github.com/artsy/metaphysics/pull/311#discussion_r66639384>:
>
> > @@ -10,61 +10,72 @@ const ArtistStatusesType = new GraphQLObjectType({
> >    fields: {
> >      artworks: {
> >        type: GraphQLBoolean,
> > +      resolve: ({ published_artworks_count }) => !!published_artworks_count,
>
> what's it mean for !! on an integer? Should it just be published_artworks_count
> > 0 (since it can never be nil)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/artsy/metaphysics/pull/311/files/efa6da403c6869ecd3fcf79a6c796d2d181c6920#r66639384>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADBlTjvZe77Yz5G5z5o5O_h65y1Yjmw0ks5qKYvEgaJpZM4IzFFy>
> .
>



-- 

[A.]


Laura Brown

Artsy <http://artsy.net/>

laura@artsy.net <caroline@artsy.net>

mzikherman commented on schema/artist/statuses.js:
```diff
@@ -10,61 +10,72 @@ const ArtistStatusesType = new GraphQLObjectType({
+          artist_id: id,
+          displayable: true,
+          size: 0,
+        }).then(count => !!count);
```
Same, should this just be `count > 0`?

mzikherman commented on schema/artist/statuses.js:
```diff
@@ -10,61 +10,72 @@ const ArtistStatusesType = new GraphQLObjectType({
       type: GraphQLBoolean,
-      resolve: ({ display_auction_link }) => display_auction_link,
+      resolve: ({ display_auction_link, hide_auction_link }) => {
+        return display_auction_link && !hide_auction_link;
```
Nice

mzikherman commented on schema/fields/markdown/format_markdown_value.js:
```diff
@@ -0,0 +1,19 @@
+      breaks: true,
+      pedantic: false,
+      sanitize: false,
+      smartypants: false,
```
👖

mzikherman commented on schema/partner_artist.js:
```diff
@@ -0,0 +1,72 @@
+    },
+  },
+  resolve: (root, { partner_id, artist_id }) =>
+    gravity(`partner/${partner_id}/artist/${artist_id}/`),
```
extra '/' at the end

mzikherman commented on schema/partner_artist.js:
```diff
@@ -0,0 +1,72 @@
+      type: GraphQLString,
+    },
+    is_display_on_partner_profile: {
+      type: GraphQLBoolean,
```
Don't these need a `resolve` with `display_on_partner_profile`? That's the JSON that comes back from gravity.

1aurabrown replied:
huh? this isn't on the partner profile page. all we want is the bio. there
is a 'use default biography' flag which is specific to the partner profile
page (id the partner has supplied a bio but for some reason ticked the box
to display our bio on their profile instead of their own. apparently this
is specific to that page and doesn't apply to the artist page.

On Fri, Jun 10, 2016 at 6:12 PM, mzikherman <notifications@github.com>
wrote:

> In schema/partner_artist.js
> <https://github.com/artsy/metaphysics/pull/311#discussion_r66640171>:
>
> > +      type: new GraphQLObjectType({
> > +        name: 'PartnerArtistCounts',
> > +        fields: {
> > +          artworks: numeral(({ published_artworks_count }) =>
> > +            published_artworks_count),
> > +          for_sale_artworks: numeral(({ published_for_sale_artworks_count }) =>
> > +            published_for_sale_artworks_count),
> > +        },
> > +      }),
> > +      resolve: (partner_artist) => partner_artist,
> > +    },
> > +    id: {
> > +      type: GraphQLString,
> > +    },
> > +    is_display_on_partner_profile: {
> > +      type: GraphQLBoolean,
>
> Don't these need a resolve with display_on_partner_profile? That's the
> JSON that comes back from gravity.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/artsy/metaphysics/pull/311/files/efa6da403c6869ecd3fcf79a6c796d2d181c6920#r66640171>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADBlTjjZsFCMdSdqJk7G82eNZbb97RRWks5qKYztgaJpZM4IzFFy>
> .
>



-- 

[A.]


Laura Brown

Artsy <http://artsy.net/>

laura@artsy.net <caroline@artsy.net>

mzikherman replied:
I'm saying `is_display_on_partner_profile` isn't a thing anywhere, the JSON response from gravity is `display_on_partner_profile`.

If you want to follow the `is_` pattern for making these fields available in the schema, don't you need to resolve those with `display_on_partner_profile`?

Something like:

```js
is_display_on_partner_profile: {
  type: GraphQLBoolean,
   resolve: ({ display_on_partner_profile }) => display_on_partner_profile,
},
```

Right now theres no `resolve`, so wouldn't this just return null since the field (as named with the `is_` prefix) doesn't exist in the JSON response for a partner artist?

dzucconi replied:
Yep

1aurabrown replied:
i misunderstood, fixed now

dzucconi commented on schema/fields/markdown.js:
```diff
@@ -1,32 +0,0 @@
-
-    return value;
-  },
-});
```
It's pretty simple to have multiple exports from a single file

1aurabrown replied:
okiedoke

dzucconi commented on schema/artwork/index.js:
```diff
@@ -10,7 +10,7 @@ import {
 } from '../../lib/helpers';
 import cached from '../fields/cached';
-import markdown from '../fields/markdown';
+import markdown from '../fields/markdown/markdown';
```
`import { markdown } from '../fields/markdown'`

1aurabrown commented:
whoever wants to look at this @mzikherman @broskoski, not sure who is less busy but i updated

broskoski commented on schema/partner_show.js:
```diff
@@ -9,7 +9,7 @@ import total from '../lib/loaders/total';
 import cached from './fields/cached';
 import date from './fields/date';
-import markdown from './fields/markdown';
+import { markdown } from './fields/markdown';
```
You'll have to do this in other places that require markdown now that there is no default export in that file. The other option is to export the original function as default (like it was before) and do something like:

```
import markdown, { formatMarkdownValue } from '../fields/markdown';
```

1aurabrown replied:
i did do this in the other files

1aurabrown commented:
@broskoski double checked, updated the markdown requirement everwhere. did what we discussed in the other pr basically, made a new field with this fallback and text and credit fields, and added a depreciation reason to blurb

broskoski commented:
✅
